### PR TITLE
Add "What is configuration management" page

### PR DESCRIPTION
**Pull request by scottslowe**
## Description

This PR adds a "What is" page that defines and discusses configuration management.

## Checklist:

- [ ] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [ ] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [ ] I have manually confirmed that all new links work.
- [ ] I added aliases (i.e., redirects) for all filename changes.
- [ ] If making css changes, I rebuilt the bundle.

### CODE REVIEW

File: themes/default/content/what-is/what-is-configuration-management.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+* **Complexity**: Managing configurations for complex systems can be intricate, leading to potential challenges in defining and maintaining configurations accurately.
+* **Overhead**: Implementing configuration management may introduce some overhead, especially in the initial setup and learning phases.
+
+## How does configuration management relate to infrastructure as code?
+
+Although configuration management (as a process) can apply to managing cloud infrastructure, the implementation of most configuration management tools make them less suitable for use in an infrastructure as code use case than a dedicated infrastructure as code tool like Pulumi. However, configuration management tools are great companions to an infrastructure as code tool like Pulumi, offering the ability to perform fine-grained configuration changes on cloud infrastructure provisioned via Pulumi (for example, ensuring that the operating system on a cloud instance is configured in a specific way or with a specific software package or application).
```
Comment (joeduffy) [inline themes/default/content/what-is/what-is-configuration-management.md:67]: Should we link to an example of this in action?

File: themes/default/content/what-is/what-is-configuration-management.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+Configuration management has a number of practical use cases for many organizations:
+
+* Automated configuration of new systems (typically servers, but could also apply to network devices or other systems)
+* Recovery from outages
+* Remediation of security flaws
+* Patching or updating software or operating systems
+* Deployment of software packages or applications
+
+## Are there any disadvantages to configuration management?
```
Comment (aaronkao) [inline themes/default/content/what-is/what-is-configuration-management.md:57]: i wonder if another disadvantage is its is "operations based" in that you apply a set of rules to get to desired state. at the end of the day, there can be enough drift that desired state never gets achieved. 

the alternative is just having entire environments configured from scratch and that is where you get the AMI based approach of AWS or Docker containers. wondering how we should work that into here.

File: themes/default/content/what-is/what-is-configuration-management.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+## Conclusion
+
+In the realm of technology, the effective management of software systems is crucial for stability and reliability. Configuration management plays a pivotal role in systematically handling changes, ensuring the integrity and functionality of a system. Utilizing tools like Ansible, Chef, or Puppet, organizations can streamline operations, enhance collaboration, and navigate the ever-evolving landscape of technology. These tools integrate well with dedicated infrastructure as code tools like Pulumi, allowing organizations to reap the benefits of automation through both infrastructure as code and configuration management. The combination of infrastructure as code and configuration management helps foster a robust and reliable digital infrastructure.
+
+Pulumi is open-source infrastructure as code in any programming language including TypeScript, Python, Go, C#, Java, and YAML. It supports all major clouds and over 120+ cloud and SaaS providers. [Try it out for free](/docs/get-started/).
```
[the remark this answers is not in the document]
Comment (aaronkao) [reply themes/default/content/what-is/what-is-configuration-management.md:73]: I also think the CTA shouldn't just be a Try it out for free link but rather a nice button

Perhaps the Purple CTA box below should be co-opted to be a more focused direct CTA for the page

**Review by aaronkao — COMMENTED** [review]
Most of my comments can be shipped as an incremental release later if needed.

**Comment by DadGPT** [comment]
One SEO note: "Configuration Management" is the right way to go compared to the shortened "Config Management".  Search volumes are much higher when the word is spelled out.  No changes needed, just wanted to confirm in case anyone was curious.

File: themes/default/content/what-is/what-is-configuration-management.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+Configuration management encompasses three broad aspects of managing and maintaining the desired state of the systems:
+
+1. **Identification**: Configuration management helps identify the systems that need to be reconfigured, patched, or updated in order to align with the desired state, as well as identifies the specific changes that need to be made to those systems.
+2. **Remediation**: Configuration management helps remediate the changes necessary to bring systems into alignment with the desired state. This could include applying software or security patches, installing (or uninstalling) specific software packages, or changing operating system configuration parameters.
+3. **Automation**: Configuration management helps automate both of the previous aspects (identification and remediation).
+
+## What are some common configuration management tools?
+
+There are a number of configuration management tools on the market, but here are a few of the most commonly mentioned tools:
```
Comment (desteves) [inline themes/default/content/what-is/what-is-configuration-management.md:24]: Thoughts on linking to each of the tools below?

File: themes/default/content/what-is/what-is-configuration-management.md (hunk 1 above)
Comment (scottslowe) [reply themes/default/content/what-is/what-is-configuration-management.md:67]: Good idea, done!

File: themes/default/content/what-is/what-is-configuration-management.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+* Recovery from outages
+* Remediation of security flaws
+* Patching or updating software or operating systems
+* Deployment of software packages or applications
+
+## Are there any disadvantages to configuration management?
+
+Like any practice, there are some potential disadvantages to configuration management. In most instances, these potential disadvantages can be outweighed by the benefits that configuration management brings:
+
+* **Learning curve**: Some configuration management tools may have a steep learning curve, requiring time and effort for teams to become proficient. This is especially the case for tools that leverage a DSL.
```
[the remark this answers is not in the document]
Comment (scottslowe) [reply themes/default/content/what-is/what-is-configuration-management.md:61]: I'd lean against that idea, at least for now.

File: themes/default/content/what-is/what-is-configuration-management.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (scottslowe) [reply themes/default/content/what-is/what-is-configuration-management.md:73]: I linked a blog post about Pulumi + Ansible earlier in the blog post per Joe's suggestion. Is there a different CTA you'd like me to use here?

File: themes/default/content/what-is/what-is-configuration-management.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+allow_long_title: true
+meta_desc: |
+    Learn about what configuration management is and why configuration management is instrumental in maintaining the health and consistency of software systems.
+type: what-is
+page_title: What is Configuration Management?
+---
+
+## What is configuration management?
+
+Configuration management is a systems engineering process for handling changes to a system's functionality, performance, and attributes in a way that preserves the system's integrity. Put more simply, it's the process of maintaining and controlling the desired state of a system to ensure it behaves as expected under various conditions.
```
[the remark this answers is not in the document]
Comment (scottslowe) [reply themes/default/content/what-is/what-is-configuration-management.md:12]: I've reworded this. Let me know what you think.

File: themes/default/content/what-is/what-is-configuration-management.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+* **Idempotence**: Configuration management tools ensure idempotence, meaning that applying the same configuration multiple times yields the same result, promoting reliability.
+* **Automation**: Automation is a fundamental attribute of configuration management tools, allowing the tools to execute tasks consistently and reduce the risk of human error.
+* **Stateless**: Configuration management tools typically do not maintain any state, or saved information, on the configuration status of systems being managed.
+
+## What are some benefits of configuration management?
+
+Configuration management offers a number of tangible benefits:
+
+* **Consistency**: By defining the desired state and then _identifying_ and _remediating_ the specific things do not align with the desired state, configuration management helps bring greater consistency to systems. For example, configuration management helps ensure that the same version of a particular software package is installed across a specific subset of operating systems instances, or that all operating systems instances are patched with a particular security update.
```
[the remark this answers is not in the document]
Comment (scottslowe) [reply themes/default/content/what-is/what-is-configuration-management.md:42]: Done

File: themes/default/content/what-is/what-is-configuration-management.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+* **Reliability**: Inconsistency is the bane of reliability. _Snowflake servers_ that have manual tweaks, incorrect hotfixes, or missing software updates make it harder to fix issues. Automation helps reduce inconsistency by eliminating human error and variance in manually-configured systems.
+* **Scalability:** Defined configurations that can be deployed via configuration management tool make it possible to scale systems faster, easier, and more reliably.
+* **Accountability**: Similar to infrastructure as code, configuration management enables the desired state to be checked into version control. Changes to the desired state can be tracked, providing accountability and visibility.
```
[the remark this answers is not in the document]
Comment (scottslowe) [reply themes/default/content/what-is/what-is-configuration-management.md:45]: Done

File: themes/default/content/what-is/what-is-configuration-management.md (hunk 2 above)
Comment (scottslowe) [reply themes/default/content/what-is/what-is-configuration-management.md:57]: I've added some text immediately before this heading (around line 56-57) mentioning immutable infrastructure (AMI-based approach) and containers. Good call.

File: themes/default/content/what-is/what-is-configuration-management.md (hunk 4 above)
Comment (scottslowe) [reply themes/default/content/what-is/what-is-configuration-management.md:24]: Seems reasonable, done

**Comment by desteves** [comment]
> http://pulumi-hugo-origin-pr-3619-e48116cd.s3-website.us-west-2.amazonaws.com

http://pulumi-hugo-origin-pr-3619-e48116cd.s3-website.us-west-2.amazonaws.com/what-is/what-is-configuration-management/ LGTM

**Review by desteves — APPROVED** [review]

**Pull request merged**